### PR TITLE
TN-1586 allow admin staff to toggle the inclusion of test accounts in

### PR DESCRIPTION
**Pull request by achen2401**
https://jira.movember.com/browse/TN-1586
adding UI checkbox to allow admin user to toggle inclusion of test accounts
- UI accessible only to Admin staff only
- test accounts are excluded by default and will be included only requested (via post)
- styling fixes
- refactor admin JS code to allow role based events
@pbugni Paul, it will be great if you can review the changes  (especially the backend ones) I had made

### CODE REVIEW

File: portal/views/staff.py [hunk 1]
```diff
@@ -147,12 +152,15 @@ def staff_index():
...
-    staff_list = staff_list.union(org_staff).all()
 
-    # only show test users to admins
-    if not user.has_role(ROLE.ADMIN.value):
-        staff_list = [s for s in staff_list if not s.has_role(ROLE.TEST.value)]
+    include_test_roles = request.form.get('include_test_roles')
+    # not including test accounts by default, unless requested
+    if not include_test_roles:
+        org_staff = org_staff.filter(
+            ~User.roles.any(Role.name == ROLE.TEST.value))
```
Comment (ivan-c) [inline portal/views/staff.py]: :1st_place_medal:

File: portal/views/staff.py [hunk 2]
```diff
@@ -147,12 +152,15 @@ def staff_index():
     ).join(UserOrganization).filter(
         and_(UserOrganization.user_id == User.id,
              UserOrganization.organization_id.in_(org_list)))
-    staff_list = staff_list.union(org_staff).all()
 
-    # only show test users to admins
-    if not user.has_role(ROLE.ADMIN.value):
-        staff_list = [s for s in staff_list if not s.has_role(ROLE.TEST.value)]
+    include_test_roles = request.form.get('include_test_roles')
```
Comment (ivan-c) [inline portal/views/staff.py]: Does this option need to be passed by POST?

It's usually easier to work with (debug/bookmark/reproduce ) GET requests and querystring parameters

Comment (achen2401) [reply portal/views/staff.py]: thanks, ivan, fixed

File: portal/views/staff.py [hunk 3]
```diff
@@ -98,7 +98,7 @@ def staff_profile(user_id):
         consent_agreements=consent_agreements)
 
 
-@staff.route('/staff', methods=('GET', 'POST'))
```
Comment (ivan-c) [inline portal/views/staff.py]: Thank you!!

**Pull request merged**